test/framework/src/main/java/org/elasticsearch/index/engine/TranslogHandler.java

```diff
@@ -60,9 +61,13 @@ long appliedOperations() {
     }
 
     public TranslogHandler(NamedXContentRegistry xContentRegistry, IndexSettings indexSettings) {
-        NamedAnalyzer defaultAnalyzer = new NamedAnalyzer("default", AnalyzerScope.INDEX, new StandardAnalyzer());
-        IndexAnalyzers indexAnalyzers =
-                new IndexAnalyzers(indexSettings, defaultAnalyzer, defaultAnalyzer, defaultAnalyzer, emptyMap(), emptyMap(), emptyMap());
+        Map<String, NamedAnalyzer> analyzers = new HashMap<>();
+        analyzers.put(AnalysisRegistry.DEFAULT_ANALYZER_NAME, new NamedAnalyzer("default", AnalyzerScope.INDEX, new StandardAnalyzer()));
+        analyzers.put(AnalysisRegistry.DEFAULT_SEARCH_ANALYZER_NAME,
+                new NamedAnalyzer("default", AnalyzerScope.INDEX, new StandardAnalyzer()));
+        analyzers.put(AnalysisRegistry.DEFAULT_SEARCH_QUOTED_ANALYZER_NAME,
+                new NamedAnalyzer("default", AnalyzerScope.INDEX, new StandardAnalyzer()));
+        IndexAnalyzers indexAnalyzers = new IndexAnalyzers(indexSettings, analyzers, emptyMap(), emptyMap());
```

Comment:
Rather than always requiring all three defaults to be contained in the Map, maybe it would be nicer to have some fallback logic in `getDefaultSearchAnalyzer` and `getDefaultSearchQuoteAnalyzer` that returns the default index analyzer if nothing specific has been added?

It would also be helpful to have an IndexAnalyzers constructor that takes a single analyzer and returns that as the default.

Reply:
niit: You only need `default` here now